### PR TITLE
Release v19.0.0

### LINKED ISSUE #456: 🗣[Discussion] Distribute individual bundles for each component
It would be awesome if I could import only a single component at a time by going a bit deeper into the filesystem, for example something like:

```
import Flex from '@primer/components/dist/Flex';
// or maybe even
import Flex from '@primer/components/Flex';
```

That mainly requires creating a JS file per component that has a `module.exports` instead of a single `index.esm.js` file. That would allow my app to only pull in the code that I am really using, instead of all components, and also solve #455 and #454.

### LINKED ISSUE #510: 🌀[API] Adjust BorderBox Props
- Remove default border radius
- Explore variants for common border types
- Make editing one of the borders easier

### LINKED ISSUE #797: unexpected token `<` when importing individual modules
importing this way doesn't work:

`import Breadcrumb from '@primer/components/src/Breadcrumbs';`

but importing from the main module works:

`import {Breadcrumb} from '@primer/components';`

This is happening for any module I import, not just breadcrumb. What could be the reason?

By the way,  I'm using create-react-app in my project.

Here's the full error I'm getting. 
```
Failed to compile
./node_modules/@primer/components/src/Breadcrumbs.js
SyntaxError: D:\VS-Code-Projects\covid19india\node_modules\@primer\components\src\Breadcrumbs.js: Unexpected token (34:64)

  32 | const BreadcrumbBase = ({className, children, theme, ...rest}) => {
  33 |   const classes = classnames(className, 'Breadcrumb')
> 34 |   const wrappedChildren = React.Children.map(children, child => <Wrapper theme={theme}>{child}</Wrapper>)
     |                                                                 ^
  35 |   return (
  36 |     <nav className={classes} aria-label="breadcrumb" {...rest}>
  37 |       <Box as="ol" my={0} pl={0}>
This error occurred during the build time and cannot be dismissed.
```

### LINKED ISSUE #801: Module not found: error when using <Link>
code example:
```
import Link from "@primer/components/src/Link";
import React from "react";

interface ExternalLinkProps {
  src: string;
  text: string;
  newTab?: boolean;
}

export const ExternalLink = (props: ExternalLinkProps) => {
  const target: string = props.newTab ? `_blank` : `_self`;
  const rel: string | undefined = props.newTab
    ? `noreferrer noopener`
    : undefined;

  return (
    <Link href={props.src} target={target} rel={rel}/>
  );
};
```
While trying to run `npm start` in console, it won't render, but show the following error:
`Module not found: Can't resolve './utils/elementType' in '{my path}/node_modules/@primer/components/src'`

### PR DESCRIPTION
**Pull request by BinaryMuse**
# Expected changes

- [x] New `BorderBox` API (breaking change)
- [x] `sx` prop + docs
- [ ] ~Removal of StyledOcticon in favor of @primer/styled-octicons~
- [x] Flash Refresh #777 
- [x] Docs updates #786 
- [x] Compilation pipeline updates #789 
- [x] Changes to `StateLabel`
- [x] Right-align for `SelectMenu.Modal`

References:

Closes #797
Closes #456
Closes #797
Closes #801
Closes #510
Closes #752 

# Release notes

## 🚨 Breaking Changes

- [x] Removed "Segoe UI Symbol" from font stack as it is not compatible with Chinese characters #769 @skw

- [x] `BorderBox` has had its default `border` prop split into `borderWidth` and `borderStyle`:

    before:

    ```javascript
    BorderBox.defaultProps = {
      border: '1px solid',
      borderColor: 'gray.2',
      borderRadius: 2
    }
    ```

    after:

    ```javascript
    BorderBox.defaultProps = {
      borderWidth: '1px',
      borderStyle: 'solid',
      borderColor: 'gray.2',
      borderRadius: 2
    }
    ```

    While this change was made to fix bugs in the ability to modify `BorderBox`'s border, it is also a breaking change if you use `border` or `border<Direction>` shorthand props to modify the border, as these will no longer work. Instead, use non-shorthand props. For example, change

    ```javascript
    <BorderBox border={0} borderTop={1}>...</BorderBox>
    ```

    to

    ```javascript
    <BorderBox borderWidth={0} borderTopWidth={1}>...</BorderBox>
    ```

     #761 @BinaryMuse 

- [x] `Flex` now behaves exactly like `Box` except that it has its `display` prop set to `flex` by default  #761 @BinaryMuse  

- [x] `Flex.Item`, previously deprecated, has been removed in favor of `Box`, which now has all the same props  #761 @BinaryMuse 

- [x] `theme.borders` has been removed; use the new `borderWidths` theme values for `borderWidth` properties (`0` is a width of `0`, `1` is a width of `1px`)  #761 @BinaryMuse 

## 🚫 Deprecations

Starting this version, we will be adding a deprecation period for breaking API changes whenever possible. When using a deprecated feature, you will see a warning in the JavaScript console that details how you should change your usage as well as the version at which we plan to remove the deprecated feature.

- [x] The `Flash` component's `scheme` prop has been deprecated in favor of a new `variant` prop. We plan to remove the `scheme` prop in version 20.

    To migrate:

    Change `scheme='blue' ` to ▶️`variant='default'`
    Change `scheme='green'` to  ▶️ `variant='success'`
    Change `scheme='red'` to  ▶️ `variant='danger'`
    Change `scheme='yellow'` to ▶️ `variant='warning'`

    #777 @emplums 

- [x] `StateLabel`'s `small` prop is deprecated in favor of a new `variant` prop. Use `variant="small"` to achieve the same result. We plan to remove the `small` prop in version 20. #798 #752 @tiaanduplessis @BinaryMuse 

## 💅 Enhancements

- [x] `Box` (and components that inherit from it, like `BorderBox`) now have access to `FLEX` system props #761 @BinaryMuse 

- [x] `Position` and its family members (`Relative`, `Absolute`, `Sticky`, and `Fixed`) now inherit from `Box` and receive `FLEX` system props.

- [x] A new prop, `sx`, has been added to all components to allow theme-aware ad-hoc styles. See [the prop's documentation](https://primer.style/components/overriding-styles) for more information. #755 @BinaryMuse 

- [x] Each component can now be imported individually, without importing all of Primer Components. For example, `import Box from '@primer/components/lib/Box'`. Users using an ESM-compatible module bundler should not use this import style. See [the usage docs](https://primer.style/components/getting-started) for more information. #789 @BinaryMuse 

- [x] The following colors have been added:

    * border.blackFade
    * border.blue
    * border.blueLight
    * border.grayDarker
    * border.green
    * border.greenLight
    * border.purple
    * border.red
    * border.redLight
    * border.white
    * border.whiteFade
    * border.yellow

     #761 @BinaryMuse 

- [x] Allow `SelectMenu.Item` to be `a` or `button` #787 

- [x] `SelectMenu.Modal` has a new property `align`; set `align='right'` to right-align the modal @dmarcey #800 

## 🐛 Bug Fixes

- [x] Typo fix #783 @andrewhillcode  
- [x] Add `text-align` to buttons #772 @albingroen

### CODE REVIEW

**Comment by emplums** [comment]
Might make sense to deprecate StyledOcticon in this release as well, Cole and I would need to create the `@primer/styled-octicons` package first though, not sure if this release has any urgency, if so we could do it in a different major version

**Comment by BinaryMuse** [comment]
> not sure if this release has any urgency

Nope, no urgency, just wanted to have a base branch to set new PRs to :)

**Comment by dmarcey** [comment]
Bundle size explorations (note all sizes are minified _and_ compressed)

1. `git clone git@github.com:dmarcey/tree-shaking-test.git`
2. `cd tree-shaking-test`
3. `yarn`
4. `yarn build`
Bundle size is `39.39 KB`
5. Uncomment lines 5 and 28 in `App.tsx` to include a `Button`.
6. `yarn build`
Bundle size is `109.38 KB`
7. `yarn add @primer/components@19.0.0-rc.d4874f0`
8. `yarn build`
Bundle size is `119.87 KB`

Size jumped ~ 10KB minified, gzipped between `18.0.0` and `19.0.0-rc.d4874f0`

**Comment by BinaryMuse** [comment]
Since the Octicon changes (most likely) won't be going out in this version, this is now officially ready for review.

File: index.d.ts [hunk 1]
```diff
@@ -369,7 +366,8 @@ declare module '@primer/components' {
 
   export interface StateLabelProps extends CommonProps {
     small?: boolean
-    status: 'issueOpened' | 'issueClosed' | 'pullOpened' | 'pullClosed' | 'pullMerged'
+    variant?: 'small' | 'normal'
+    status: 'issueOpened' | 'issueClosed' | 'pullOpened' | 'pullClosed' | 'pullMerged' | 'draft'
```
Comment (emplums) [inline index.d.ts:370]: For whatever reason, testing these in the CodeSandbox keeps getting me a TS error for both `variant` and `status` 🤔 

https://codesandbox.io/s/primer-components-ts-testing-47dbw?file=/src/index.tsx:783-844

**Comment by emplums** [comment]
Testing sandbox: https://codesandbox.io/s/primer-components-ts-testing-47dbw

**Pull request merged**